pinot-query-planner/src/main/java/org/apache/pinot/query/planner/logical/RelToStageConverter.java

```diff
@@ -115,8 +115,8 @@ private static StageNode convertLogicalJoin(LogicalJoin node, int currentStageId
     RexCall joinCondition = (RexCall) node.getCondition();
 
     // Parse out all equality JOIN conditions
-    int leftNodeOffset = node.getLeft().getRowType().getFieldList().size();
-    List<List<Integer>> predicateColumns = PlannerUtils.getJoinKeyFromConditions(joinCondition, leftNodeOffset);
+    List<List<Integer>> predicateColumns =
+        PlannerUtils.getJoinKeyFromConditions(joinCondition, node.analyzeCondition());
 
     FieldSelectionKeySelector leftFieldSelectionKeySelector = new FieldSelectionKeySelector(predicateColumns.get(0));
     FieldSelectionKeySelector rightFieldSelectionKeySelector = new FieldSelectionKeySelector(predicateColumns.get(1));
```

Comment:
```suggestion
    JoinInfo joinInfo = node.analyzeCondition();
    FieldSelectionKeySelector leftFieldSelectionKeySelector = new FieldSelectionKeySelector(joinInfo.leftKeys);
    FieldSelectionKeySelector rightFieldSelectionKeySelector = new FieldSelectionKeySelector(joinInfo.rightKeys);
```

Reply:
in addition you can also assert that `joinInfo.nonEquiConditions` is empty. we can later support inequality join using this.

Reply:
Done.